cpp/src/lists/utilities.cpp

```diff
@@ -28,18 +28,11 @@ namespace detail {
  * @param[in] stream The cuda-stream to synchronize on, when reading from device memory
  * @return cudf::size_type The number of child rows in the lists column
  */
-static cudf::size_type get_num_child_rows(cudf::column_view const& list_offsets,
-                                          rmm::cuda_stream_view stream)
+cudf::size_type get_num_child_rows(cudf::column_view const& list_offsets,
+                                   rmm::cuda_stream_view stream)
 {
   // Number of rows in child-column == last offset value.
-  cudf::size_type num_child_rows{};
-  CUDA_TRY(cudaMemcpyAsync(&num_child_rows,
-                           list_offsets.data<cudf::size_type>() + list_offsets.size() - 1,
-                           sizeof(cudf::size_type),
-                           cudaMemcpyDeviceToHost,
-                           stream.value()));
-  stream.synchronize();
-  return num_child_rows;
+  return get_value<size_type>(list_offsets, list_offsets.size() - 1, stream);
```

Comment:
This gets the last entry in the offsets column. Is this always the size? What happens if the list column itself has an offset?

Reply:
Hmm... It turns out that in this case, the list column is being constructed (in `rolling_detail.cuh`). So it doesn't have an offset.

But in the general case, we'd need to subtract `get_value<size_type>(list_offsets, list.offset(), stream)`, if the offset is non-zero?
Is that how this sort of thing is handled for `string_view` columns?

Reply:
It is very rare we need the number of bytes in the chars child column from within a sliced column. But yes, when we need that we subtract the two offset values to get the number of bytes.
At the very least, I would just change the name/description to what it is actually doing.

Right now, it appears to be just a thin wrapper on `get_value()` 
So perhaps it could be eliminated altogether and just use `get_value()` where this is called instead.

Reply:
Removed. I'm now using `get_value()` directly.